source/socks5-support/socks5.rst

```diff
@@ -0,0 +1,192 @@
+==============
+SOCKS5 Support
+==============
+
+:Spec Title: SOCKS5 Support
+:Spec Version: 1.0.0
+:Author: Anna Henningsen
+:Advisors: TODO
+:Status: Draft
+:Type: Standards
+:Minimum Server Version: N/A
+:Last Modified: 2021-11-19
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+SOCKS5 is a standardized protocol for connecting to network services through
+a separate proxy server. It can be used for connecting to hosts that would
+otherwise be unreachable from the local network by connecting to a proxy
+server, which receives the intended target host’s address from the client
+and then connects to that address.
+
+This specification defines driver behaviour when connecting to MongoDB services
+through a SOCKS5 proxy.
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in `RFC 2119 <https://www.ietf.org/rfc/rfc2119.txt>`__.
+
+Specification
+=============
+
+
+Terms
+-----
+
+SOCKS5
+^^^^^^
+
+The SOCKS protocol, version 5, as defined in `RFC1928 <https://datatracker.ietf.org/doc/html/rfc1928>`__,
+restricted to either no authentication or username/password authentication
+as defined in `RFC1929 <https://datatracker.ietf.org/doc/html/rfc1929>`__.
+
+
+MongoClient Configuration
+-------------------------
+
+proxyHost
+^^^^^^^^^
+
+To specify to the driver to connect using a SOCKS5 proxy, a connection string
+option of :code:`proxyHost=host` MUST be added to the connection string
+or passed through an equivalent :code:`MongoClient` option.
+This option specifies a domain name or IPv4 or IPv6 address on which
```

Comment:
@nbbeeken Do you happen to have an example of that? Also, the current set of invalid tests is invalid regardless of whether the driver supports Socks5 or not, but in the scope we said we’d try to keep the new (passing) tests somewhat separate so that the other drivers won’t have issues updating the spec tests until they start implementing Socks5 support?

Reply:
I think I take back the suggestion, this is just general URI parsing behavior, its not even specific to IPv6, any searchParam can be precent escaped so it shouldn't need special handling I'd imagine.
And that makes sense about holding off on valid cases (or placing them elsewhere) I know our driver will handle this automagicially so there isn't intervention needed in the implementation.